docs/hacking/install.rst

```diff
@@ -186,64 +171,139 @@ system that Docker can be installed on:
+
+
+.. _activating_your_virtual_environment:
+
+Activating your virtual environment
+-----------------------------------
+
+Activate the virtual environment that you've created:
+
+.. code-block:: bash
+
+   source .venv/bin/activate
+
+.. tip::
+
+   You'll need to re-activate this virtualenv with the
+   ``source .venv/bin/activate`` command each time you open a new terminal,
+   before running h.
+   See the `Virtual Environments`_ section in the Hitchhiker's guide to
+   Python for an introduction to Python virtual environments.
+
+.. _Virtual Environments: http://docs.python-guide.org/en/latest/dev/virtualenvs/
+
+
+Running h
+---------
+
+Start the Hypothesis web service:
 
 .. code-block:: bash
 
     make dev
 
+The first time you run ``make dev`` it might take a while to start because
+it'll need to install the application dependencies and build the client assets.
+
 This will start the server on port 5000 (http://localhost:5000), reload the
 application whenever changes are made to the source code, and restart it should
 it crash for some reason.
 
-.. note::
-    Using the bookmarklet or otherwise embedding the application may not
-    be possible on sites accessed via HTTPS due to browser policy restricting
-    the inclusion of non-SSL content.
-
 If you are making changes to the client, or the JavaScript code or styles for the
-service, you may find it useful to install Gulp_ and run the `watch` task. This
-will automatically rebuild the assets whenever the source files change.
+service, you may find it useful to run this command from your ``h`` directory
+in a separate terminal:
 
 .. code-block:: bash
 
-    npm install -g gulp-cli
     gulp watch
 
-When `gulp watch` is running, you can visit http://localhost:3000
+``gulp watch`` will automatically rebuild the assets whenever the source files
+change.
+
+When ``gulp watch`` is running, you can visit http://localhost:3000
 to see a page with an embedded Hypothesis client which will automatically reload
 when styles, templates or JavaScript source files are changed.
 
 .. _Gulp: http://gulpjs.com/
 
-If you are running background tasks, you must have a Celery worker running. To
-run a Celery worker in development, run:
+
+Running the WebSocket for real-time features
+--------------------------------------------
+
+To test h's real-time features (for example, seeing new or updated annotations
+from other users appear in your browser without a page reload) you need to run
+a websocket process.  Open a new terminal, ``cd`` into your ``h`` directory,
+:ref:`activate your Python virtual environment <activating_your_virtual_environment>`,
+then run:
+
+.. code-block:: bash
+
+   gunicorn --reload --paste conf/development-websocket.ini
+
+
+Running Celery for background tasks
+-----------------------------------
+
+To test h's background tasks (for example, sending emails) you need to run a
+Celery worker process.
+Open a new terminal, ``cd`` into your ``h`` directory,
```

Comment:
What I have gone for here (and in general on this page) is to keep the docs as short and concise as possible (one sentence in this case), linking to external resources for further reading where relevant, while also making it possible to follow the docs without any prior knowledge.

Of course "without prior knowledge" is open to interpretation so I don't explain what a computer is or anything.

But this sentence is making it clear that hypothesis-celery is a separate command that needs to be run in a separate terminal, from the h dir, and with the venv activated. Again I think the burden to maintaining this is low and it should be helpful to people who aren't Python devs